installer/pkg/components/gitpod/cronjob.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package gitpod
+
+import (
+	"crypto/sha512"
+	"fmt"
+
+	"github.com/gitpod-io/gitpod/installer/pkg/common"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+)
+
+func cronjob(ctx *common.RenderContext) ([]runtime.Object, error) {
+	installationTelemetryComponent := fmt.Sprintf("%s-telemetry", Component)
+
+	objectMeta := metav1.ObjectMeta{
+		Name:      installationTelemetryComponent,
+		Namespace: ctx.Namespace,
+		Labels:    common.DefaultLabels(Component),
+	}
+
+	return []runtime.Object{
+		// Telemetry calls home to provide anonymous information on the installation
+		&batchv1.CronJob{
+			TypeMeta:   common.TypeMetaBatchCronJob,
+			ObjectMeta: objectMeta,
+			Spec: batchv1.CronJobSpec{
+				Schedule:                   TelemetryCronSchedule,
+				SuccessfulJobsHistoryLimit: pointer.Int32(3),
+				FailedJobsHistoryLimit:     pointer.Int32(1),
+				ConcurrencyPolicy:          batchv1.ReplaceConcurrent,
+				JobTemplate: batchv1.JobTemplateSpec{
+					ObjectMeta: objectMeta,
+					Spec: batchv1.JobSpec{
+						Template: v1.PodTemplateSpec{
+							ObjectMeta: objectMeta,
+							Spec: v1.PodSpec{
+								RestartPolicy:      v1.RestartPolicyOnFailure,
+								ServiceAccountName: Component,
+								EnableServiceLinks: pointer.Bool(false),
+								// The init container is designed to emulate Helm hooks
+								InitContainers: []v1.Container{*common.DatabaseWaiterContainer(ctx)},
```

Comment:
Is this a left-over from copying this or is it really intended to wait for the database?

Reply:
Me? Copy paste? How very dare you 😆 

It's actually in there intentionally as the next iteration will make a DB call to get whether the user wants to send the telemetry data. Using the `InitContainers` is a pattern used elsewhere in the Installer to ensure that a DB connection can be made. Arguably, it's of little practical value in the telemetry as only sending once per day, but I felt it good practice to follow the established pattern - I think it should stay, but if you want it removed I won't put up much resistance to it.

Reply:
> Me? Copy paste? How very dare you :laughing: 

Ha Ha! :innocent: 

Fair enough. A dependency on the database sounds a little odd to me at this point but why not. Let's keep it like this for now.